molo/surveys/rules.py

```diff
@@ -402,9 +441,12 @@ def clean(self):
 
     def test_user(self, request, user=None):
         if user:
-            # This rule currently does not support testing a user directly
-            # TODO: Make this test a user directly when the rule uses
-            # historical data
+            # Create a fake request so we can use the adapter
+            request = RequestFactory().get('/')
```

Comment:
I dislike this, since it's using a class intended for tests, but it's how it's done in `wagtail_personalisation` when testing users against static segments to populate them at creation https://github.com/praekeltfoundation/wagtail-personalisation/blob/develop/src/wagtail_personalisation/forms.py#L105

Reply:
I know what you mean, this is kind of weird but I don't think it's too terrible :)

As we discussed it might be possible test that the adapter is one that we know to be supported and return early / raise an error if it's not? But also I'm happy with this as it stands.

Reply:
Yeah, I came up with something I'm happier with in the wagtail_personalisation code. For this repo I think it's ok for it to return 0 for the case where we pass in a user. Largely because this is the project that adds the PersistentSegmentAdapter.